tests/Unit/Domain/Test_SurfaceElement.cpp

```diff
@@ -0,0 +1,129 @@
+
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Determinant.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/CoordinateMaps/Affine.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.tpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.tpp"
+#include "Domain/CoordinateMaps/Rotation.hpp"
+#include "Domain/CoordinateMaps/Tags.hpp"
+#include "Domain/CoordinateMaps/Wedge.hpp"
+#include "Domain/CreateInitialElement.hpp"
+#include "Domain/Creators/Shell.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Domain/InterfaceLogicalCoordinates.hpp"
+#include "Domain/Structure/ExcisionSphere.hpp"
+#include "Domain/Structure/InitialElementIds.hpp"
+#include "Domain/Structure/OrientationMap.hpp"
+#include "Domain/SurfaceElement.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "NumericalAlgorithms/LinearOperators/DefiniteIntegral.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace domain {
+namespace {
+
+// Checks the euclidean surface element against the magnitude of the
+// unnormalized normal vector.
+template <typename TargetFrame>
+void test_surface_element_against_normal_vector() {
+  const auto element_map_3d = ElementMap<3, TargetFrame>(
+      ElementId<3>(0),
+      make_coordinate_map_base<Frame::BlockLogical, TargetFrame>(
+          CoordinateMaps::Wedge(1., 2., 1., 1., OrientationMap<3>{}, true)));
+
+  const Mesh<2> interface_mesh{8, Spectral::Basis::Legendre,
+                               Spectral::Quadrature::GaussLobatto};
+  const Direction<3> direction{};
+  const auto coords = interface_logical_coordinates(interface_mesh, direction);
+  const auto inverse_jacobian = element_map_3d.inv_jacobian(coords);
+  const auto det_volume = determinant(inverse_jacobian);
+  const auto result_1 =
+      euclidean_surface_element(inverse_jacobian, det_volume, direction);
+  const auto result_2 = euclidean_surface_element(inverse_jacobian, direction);
+  const auto face_normal =
+      unnormalized_face_normal(interface_mesh, element_map_3d, direction);
+  const auto face_normal_mag = magnitude(face_normal);
+  const DataVector face_normal_mag_corrected =
+      face_normal_mag.get() / det_volume.get();
+  CHECK_ITERABLE_APPROX(result_1.get(), result_2.get());
+  CHECK_ITERABLE_APPROX(face_normal_mag_corrected, result_1.get());
+}
+
+// A test function sin^2(theta) * cos^2(phi) which evaluates to 4/3 * pi * R^2
+// when integrated over a sphere with radius R.
+DataVector test_function(const tnsr::I<DataVector, 3>& inertial_coords) {
+  const DataVector theta =
+      atan2(hypot(inertial_coords.get(0), inertial_coords.get(1)),
+            inertial_coords.get(2));
```

Comment:
reminder to prefer `get<0>(coords)` over `coords.get(0)` when you know the indices at compile time